src/common/image.c

```diff
@@ -878,15 +919,19 @@ static uint32_t dt_image_import_internal(const int32_t film_id, const char *file
     flags |= DT_IMAGE_HAS_TXT;
     g_free(extra_file);
   }
+
+  const int64_t new_image_position = createNextImagePosition();
```

Comment:
Generating the next image position like that isn't thread safe. When adding several images in parallel it might break. Would it be possible to use a subquery to determine the right value on the fly?

Reply:
Done 

I thought a while about this and found that this here is possible:

    INSERT INTO main.images (id, film_id, filename, caption, description, license, sha1sum, flags, version, 
        max_version, history_end, position) 
    SELECT NULL, ?1, ?2, '', '', '', '', ?3, 0, 0, 0, (IFNULL(MAX(position),0) & (4294967295 << 32))  + (1 << 32)
    FROM images